remote-api/src/main/java/org/alfresco/rest/api/impl/RenditionsImpl.java

```diff
@@ -197,13 +196,12 @@ public void init()
             }
         }
 
-        List<ChildAssociationRef> nodeRefRenditions = renditionService2.getRenditions(validatedNodeRef);
-        if (!nodeRefRenditions.isEmpty())
+        List<RenditionContentData> renditionContentDataList = renditionService2.getRenditions(validatedNodeRef);
```

Comment:
Add ability to switch between Rendition2 & Rendition3

Reply:
The idea is to make the REST api(RenditionsImpl) layer transparent of the underlying storage of Renditions(Nodes or Properties). This would remove any tight coupling it has with Noderef. The RenditionService would return a RenditionContentData object irrespective of how renditions are stored. The changes related to renditions will be confined to the Service layer only. 
This would also avoid the need to maintain multiple RenditionService implementations. The Rendtions2ServiceImpl would be refactored to include the methods of the new Implementation(RenditionService2NewImpl) and then we can delete the new Implementation class.  This was intentionally done to enable us to do a demo with minimal set of changes.
The switch between different storage mechanisms will be based on the new property inside Rendition2ServiceImpl.

Reply:
Yup, just to add to what Nithin said, refactoring the Rendtions2ServiceImpl is one of the outstanding issues still left to do on the list in the jira- this will be quite a big task as it will affect a lot of the older code using renditions.